examples/cosmos/rollkit_test.go

```diff
@@ -0,0 +1,221 @@
+package cosmos_test
+
+import (
+	"context"
+	"cosmossdk.io/math"
+	"encoding/base64"
+	"fmt"
+	"github.com/avast/retry-go/v4"
+	cmtjson "github.com/cometbft/cometbft/libs/json"
+	"github.com/cometbft/cometbft/privval"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+	"strings"
+	"testing"
+	"time"
+)
+
+var sdk50Genesis = []cosmos.GenesisKV{
+	{
+		Key:   "app_state.gov.params.voting_period",
+		Value: votingPeriod,
+	},
+	{
+		Key:   "app_state.gov.params.max_deposit_period",
+		Value: maxDepositPeriod,
+	},
+	{
+		Key:   "app_state.gov.params.min_deposit.0.denom",
+		Value: "stake",
+	},
+	{
+		Key:   "app_state.gov.params.min_deposit.0.amount",
+		Value: "1",
+	},
+}
+
+func TestRollkitCelestiaDevnet(t *testing.T) {
+	ctx := context.Background()
+	var rollkitChain *cosmos.CosmosChain
+
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			Name:      "gm",
+			ChainName: "gm",
+			Version:   "tutorial-local-da",
+			ChainConfig: ibc.ChainConfig{
+				Type:    "cosmos",
+				Name:    "gm",
+				ChainID: "gm",
+				Images: []ibc.DockerImage{
+					{
+						Repository: "ghcr.io/gjermundgaraba/gm",
+						Version:    "tutorial-local-da",
+						UidGid:     "10001:10001",
+					},
+				},
+				Bin:              "gmd",
+				Bech32Prefix:     "gm",
+				Denom:            "stake",
+				CoinType:         "118",
+				SigningAlgorithm: "",
+				GasPrices:        "0stake",
+				GasAdjustment:    2.0,
+				TrustingPeriod:   "112h",
+				NoHostMount:      false,
+				SkipGenTx:        false,
+				PreGenesis:       nil,
+				ModifyGenesis: func(config ibc.ChainConfig, bytes []byte) ([]byte, error) {
```

Comment:
Ahh I see now. Ideally we would abstract this away and have a RollKit section which does this but does not have the user be informed of all the setup complexities. Everything here looks reasonable imo

great work!

Reply:
Yeah, it would be great to automagically sort this somehow without all the manual stuff for sure :)